cachito/workers/pkg_managers/pip.py

```diff
@@ -1514,3 +1517,101 @@ def upload_raw_package(repo_name, artifact_path, dest_dir, filename, is_request_
     to_nexus_hoster = not is_request_repository
     log.debug("Uploading %r as a raw package to the %r Nexus repository", artifact_path, repo_name)
     nexus.upload_raw_component(repo_name, dest_dir, components, to_nexus_hoster)
+
+
+def generate_pip_conf_content(nexus_repo_identifier, username, password, custom_ca_path=None):
+    """
+    Generate a pip.conf file with the nexus repo url and authentication configured.
+
+    :param str nexus_repo_identifier: the nexus repo identifier,
+        e.g. cachito-pip-hosted-<request_id>
+    :param str username: the username of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str password: the password of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str custom_ca_path: the path to set ``cert`` to in the pip.conf file; if not provided,
+        this option will be omitted
+    :return: the contents of the pip.conf file
+    :rtype: str
+    """
+    # Instead of getting the token from Nexus, use basic authentication as supported by Nexus:
+    # https://help.sonatype.com/repomanager3/formats/npm-registry#npmRegistry-AuthenticationUsingBasicAuth
```

Comment:
This links to documentation about NPM, here is the relevant part of PyPI documentation: https://help.sonatype.com/repomanager3/formats/pypi-repositories#PyPIRepositories-ConfiguringPyPIClientTools

The link above does not mention anything specific about authentication, so I'm assuming we should just treat it as regular PyPI, making this the most relevant  document: https://pip.pypa.io/en/stable/user_guide/#basic-authentication-credentials